### PR TITLE
feat: support "upload and downlod RDP file"

### LINKED ISSUE #45: feature: add "upload and downlod RDP file" feature
Added some shortcut to session for:
1-Disconnect session  /Logout
2-Upload for RDP
3-Download for RDP
4-Ctrl + C
5-Ctrl + V
6-Alt +Tab

### PR DESCRIPTION
**Pull request by tulutulupi**
Fix: https://github.com/casvisor/casvisor/issues/45

Add functionality to upload an RDP file to generate an asset and download the existing asset's RDP file.

### CODE REVIEW

**Comment by tulutulupi** [comment]
@hsluoyz ready for review

File: web/src/AssetListPage.js [hunk 1]
```diff
@@ -82,17 +83,18 @@ class AssetListPage extends BaseListPage {
       });
   };
 
-  newAsset() {
+  newAsset(ip = "127.0.0.1", username = "Administrator",
+    name = `machine_${Setting.getRandomName()}`, displayName = `New Machine - ${Setting.getRandomName()}`) {
```
Comment (hsluoyz) [inline web/src/AssetListPage.js]: Write new code, don't mess with this code

File: web/src/AssetListPage.js [hunk 2]
```diff
@@ -140,6 +141,64 @@ class AssetListPage extends BaseListPage {
...
   }
 
+  parseRdpFile = (content) => {
+    const lines = content.split(/\r?\n/);
+    let ip = "";
+    let username = "";
+
+    for (const line of lines) {
+      const trimmedLine = line.trim();
+      if (trimmedLine.startsWith("full address:s:")) {
```
Comment (hsluoyz) [inline web/src/AssetListPage.js]: Use regex, let ChatGPT write it, ChatGPT can do better

Just use asset as return value

File: web/src/AssetListPage.js [hunk 3]
```diff
@@ -140,6 +141,64 @@ class AssetListPage extends BaseListPage {
...
+      reader.readAsText(file);
+    });
+  };
+
+  handleFileRemove(file) {
+    this.uploadComponentRef.current && this.uploadComponentRef.current.onRemove(file);
+  }
+
+  renderUpload() {
+    const props = {
```
Comment (hsluoyz) [inline web/src/AssetListPage.js:202]: Refer to Casdoor similar code

File: web/src/locales/de/data.json [hunk 4]
```diff
@@ -12,17 +12,20 @@
...
     "Gatewat port": "Gatewat port",
     "Gatewat port - Tooltip": "Gatewat port - Tooltip",
+    "Invalid RDP file": "Invalid RDP file",
     "Memory": "Memory",
     "New Asset": "New Asset",
     "OS": "OS",
     "OS - Tooltip": "OS - Tooltip",
-    "Please select a asset": "Please select a asset"
+    "Please select a asset": "Please select a asset",
+    "Upload (.rdp)": "Upload (.rdp)"
```
Comment (hsluoyz) [inline web/src/locales/de/data.json]: Only translate "Upload"

File: web/src/AssetEditPage.js [hunk 5]
```diff
@@ -169,6 +169,26 @@ class AssetEditPage extends React.Component {
     return asset;
   }
 
+  downLoadRDPFile(asset) {
```
Comment (hsluoyz) [inline web/src/AssetEditPage.js]: RDP -> Rdp for everywhere except labels

File: web/src/AssetEditPage.js [hunk 6]
```diff
@@ -178,7 +198,12 @@ class AssetEditPage extends React.Component {
           {this.state.mode === "add" ? i18next.t("asset:New Asset") : i18next.t("asset:Edit Asset")}&nbsp;&nbsp;&nbsp;&nbsp;
           <Button onClick={() => this.submitAssetEdit(false)}>{i18next.t("general:Save")}</Button>
           <Button style={{marginLeft: "20px"}} type="primary" onClick={() => this.submitAssetEdit(true)}>{i18next.t("general:Save & Exit")}</Button>
-          {this.state.mode === "add" ? <Button style={{marginLeft: "20px"}} onClick={() => this.deleteAsset()}>{i18next.t("general:Cancel")}</Button> : null}
+          {(asset.type === "RDP" && this.state.mode !== "add") ?
```
Comment (hsluoyz) [inline web/src/AssetEditPage.js:202]: Don't change existing code

File: web/src/AssetEditPage.js [hunk 7]
```diff
@@ -289,7 +314,7 @@ class AssetEditPage extends React.Component {
           <Col span={22} >
             <Input
               value={asset.port}
-              defaultValue={this.getDefaultPort(asset.protocol)}
+              defaultValue={this.getDefaultPort(asset.type)}
```
Comment (hsluoyz) [inline web/src/AssetEditPage.js]: Don't change existing code

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by tulutulupi** [comment]
@hsluoyz ready for review again

File: web/src/AssetEditPage.js [hunk 8]
```diff
@@ -179,6 +199,11 @@ class AssetEditPage extends React.Component {
           <Button onClick={() => this.submitAssetEdit(false)}>{i18next.t("general:Save")}</Button>
           <Button style={{marginLeft: "20px"}} type="primary" onClick={() => this.submitAssetEdit(true)}>{i18next.t("general:Save & Exit")}</Button>
           {this.state.mode === "add" ? <Button style={{marginLeft: "20px"}} onClick={() => this.deleteAsset()}>{i18next.t("general:Cancel")}</Button> : null}
+          {(asset.type === "RDP" && this.state.mode !== "add") ?
+            <Button style={{marginLeft: "20px"}} type="primary" onClick={() => this.downLoadRdpFile(asset)}>
+              {i18next.t("asset:Download Rdp file")}
```
Comment (hsluoyz) [inline web/src/AssetEditPage.js]: UI is RDP

File: web/src/AssetListPage.js [hunk 9]
```diff
@@ -140,6 +156,62 @@ class AssetListPage extends BaseListPage {
...
+
+    const ip = ipMatch ? ipMatch[1].trim() : "";
+    const username = usernameMatch ? usernameMatch[1].trim() : "";
+    if (ip !== "" && username !== "") {
+      const asset = this.newAsset();
+      asset.endpoint = ip;
+      asset.username = username;
+      return asset;
+    } else {
+      Setting.showMessage("error", i18next.t("asset:Invalid Rdp file"));
```
Comment (hsluoyz) [inline web/src/AssetListPage.js]: UI is RDP

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by tulutulupi** [comment]
@hsluoyz ready for review again

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**